dmwmclient/datasvc.py

```diff
@@ -81,3 +81,199 @@ def __init__(self, client, datasvc_base=None, phedex_instance=None):
+        lfn              logical file name
+        """
+
+        resjson = await self.jsonmethod("errorlog", **params)
+        out = []
+        for _instance in resjson["phedex"]["link"]:
+            for _block in _instance["block"]:
+                for _file in _block["file"]:
+                    for _transfer_error in _file["transfer_error"]:
+                        out.append(
+                            {
+                                "Link": _instance["from"] + " to " + _instance["to"],
+                                "LFN": _file["name"],
+                                "file Checksum": _file["checksum"],
+                                "file size (GB)": _file["size"] / 1000000000.0,
+                                "Block name": _block["name"],
+                                "Error log": str(_transfer_error["detail_log"]["$t"]),
+                                "From PFN": _transfer_error["from_pfn"],
+                                "To PFN": _transfer_error["to_pfn"],
+                                "Time": _transfer_error["time_done"],
+                            }
+                        )
+        df = pandas.io.json.json_normalize(out)
+        format_dates(df, ["Time"])
+        return df
+
+    async def blockarrive(self, **params):
+
+        """Return estimated time of arrival for blocks currently subscribed for transfer. If the estimated time of arrival (ETA)
+        cannot be calculated, or the block will never arrive, a reason for the missing estimate is provided.
+        Parameters
+        ----------
+        id                    block id
+        block                 block name, could be multiple, could have wildcard
+        dataset               dataset name, could be multiple, could have wildcard
+        to_node               destination node, could be multiple, could have wildcard
+        priority              priority, could be multiple
+        update_since          updated since this time
+        basis                 technique used for the ETA calculation, or reason it's missing.
+        arrive_before         only show blocks that are expected to arrive before this time.
+        arrive_after          only show blocks that are expected to arrive after this time.
+
+        """
+
+        resjson = await self.jsonmethod("blockarrive", **params)
+        out = []
+        for _block in resjson["phedex"]["block"]:
+            for _destination in _block["destination"]:
+                out.append(
+                    {
+                        "Block Name": _block["name"],
+                        "Destination": _destination["name"],
+                        "Time Arrive": _destination["time_arrive"],
+                        "Time update": _destination["time_update"],
+                        "Number of files": _destination["files"],
+                        "Block size (GB)": _destination["bytes"] / 1000000000.0,
+                        "Basis code": _destination["basis"],
```

Comment:
You could provide here an extra column with the human-readable version as translated by the `BLOCKARRIVE_BASISCODE` map declared at the top of this file.